test/e2e/dra/versionskew_test.go

```diff
@@ -0,0 +1,302 @@
+		framework.ExpectNoError(e2epod.WaitForPodNotFoundInNamespace(ctx, f.ClientSet, pod.Name, namespace.Name, f.Timeouts.PodDelete))
+
+		// Create another claim and pod, this time using the latest Kubernetes.
+		claim = b.externalClaim()
+		pod = b.podExternal()
+		pod.Spec.ResourceClaims[0].ResourceClaimName = &claim.Name
+		b.create(ctx, claim, pod)
+		b.testPod(ctx, f, pod)
+
+		// Roll back.
+		tCtx = ktesting.Begin(tCtx, "downgrade")
+		cluster.Modify(tCtx, restoreOptions)
+		tCtx = ktesting.End(tCtx)
+
+		// TODO: ensure that kube-controller-manager is up-and-running.
+		// This works around https://github.com/kubernetes/kubernetes/issues/132334 and can be removed
+		// once a fix for that is backported.
+		tCtx = ktesting.Begin(tCtx, "wait for kube-controller-manager")
+		ktesting.Eventually(tCtx, func(tCtx ktesting.TContext) string {
+			return cluster.GetSystemLogs(tCtx, "kube-controller-manager")
+		}).Should(gomega.And(
+			gomega.ContainSubstring(`successfully renewed lease kube-system/kube-controller-manager`),
+			gomega.ContainSubstring(`"Caches populated" type="*v1beta1.ResourceClaim"`),
+			gomega.ContainSubstring(`"Caches populated" type="*v1.Pod"`),
+		))
+		tCtx = ktesting.End(tCtx)
+
+		// We need to clean up explicitly because the normal
+		// cleanup doesn't work (driver shuts down first).
+		//
+		// The retry loops are necessary because of a stale connection
+		// to the restarted apiserver. Sometimes, attempts fail with "EOF" as error
+		// or (even weirder) with
+		//     getting *v1.Pod: pods "tester-2" is forbidden: User "kubernetes-admin" cannot get resource "pods" in API group "" in the namespace "dra-9021"
+		ktesting.Eventually(tCtx, func(tCtx ktesting.TContext) error {
+			return f.ClientSet.ResourceV1beta1().ResourceClaims(namespace.Name).Delete(tCtx, claim.Name, metav1.DeleteOptions{})
+		}).Should(gomega.Succeed(), "delete claim after downgrade")
+		ktesting.Eventually(tCtx, func(tCtx ktesting.TContext) error {
+			return f.ClientSet.CoreV1().Pods(namespace.Name).Delete(tCtx, pod.Name, metav1.DeleteOptions{})
+		}).Should(gomega.Succeed(), "delete pod after downgrade")
+		ktesting.Eventually(tCtx, func(tCtx ktesting.TContext) *v1.Pod {
+			pod, err := f.ClientSet.CoreV1().Pods(namespace.Name).Get(tCtx, pod.Name, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				return nil
+			}
+			tCtx.ExpectNoError(err, "get pod")
+			return pod
+		}).Should(gomega.BeNil(), "no pod after deletion after downgrade")
+	})
+})
+
+// sourceVersion identifies the Kubernetes git version based on hack/print-workspace-status.sh.
+//
+// Adapted from https://github.com/kubernetes-sigs/kind/blob/3df64e784cc0ea74125b2a2e9877817418afa3af/pkg/build/nodeimage/internal/kube/source.go#L71-L104
+func sourceVersion(tCtx ktesting.TContext, kubeRoot string) (gitVersion string, dockerTag string, err error) {
+	// Get the version output.
+	cmd := exec.CommandContext(tCtx, "hack/print-workspace-status.sh")
```

Comment:
I don't think e2e.test should be doing this, or compiling anything. This is something we'd do in a tool like kubetest2 or kinder. It's a slow, orthogonal concern to testing.